keras_nlp/layers/transformer_decoder.py

```diff
@@ -223,28 +261,41 @@ def call(
         self_attended = self._add_and_norm(
             self_attended, decoder_sequence, self._decoder_attention_layernorm
         )
-        # Encoder-decoder attention.
-        encoder_decoder_attended = self._encoder_decoder_attention_layer(
-            query=self_attended,
-            value=encoder_sequence,
-            key=encoder_sequence,
-            attention_mask=encoder_mask,
-        )
-        encoder_decoder_attended = self._enc_dec_attentiondropout(
-            encoder_decoder_attended,
-        )
-        encoder_decoder_attended = self._add_and_norm(
-            encoder_decoder_attended,
-            self_attended,
-            self._enc_dec_attention_layernorm,
-        )
-        # Feedforward.
-        feed_forward_output = self._feed_forward(encoder_decoder_attended)
-        return self._add_and_norm(
-            encoder_decoder_attended,
-            feed_forward_output,
-            self._feedforward_layernorm,
-        )
+
+        if self._encoder_decoder_attention_layer is None:
```

Comment:
One minor comment. It might be nice if you rename the self_attended variable to attention_output, and do something like this.

```
attention_output = self._add_and_norm(...)

if encoder_sequence is not None:
    ... cross attention ...
    attention_output = self._add_and_norm(...)

feed_forward_output = self._feed_forward(attention_output)
return self._add_and_norm(...)
```

So basically bring this back to the single return statement. As a reader, that would make it much clearer how the computation is flowing overall with and without `encoder_sequence`.